bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/BatchedReadEntryProcessor.java

```diff
@@ -0,0 +1,97 @@
+/*
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+package org.apache.bookkeeper.proto;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.util.Recycler;
+import io.netty.util.ReferenceCounted;
+import java.util.concurrent.ExecutorService;
+import org.apache.bookkeeper.proto.BookieProtocol.BatchedReadRequest;
+import org.apache.bookkeeper.util.ByteBufList;
+
+class BatchedReadEntryProcessor extends ReadEntryProcessor {
+
+    public static BatchedReadEntryProcessor create(BatchedReadRequest request,
+                                            BookieRequestHandler requestHandler,
+                                            BookieRequestProcessor requestProcessor,
+                                            ExecutorService fenceThreadPool,
+                                            boolean throttleReadResponses) {
+        BatchedReadEntryProcessor rep = RECYCLER.get();
+        rep.init(request, requestHandler, requestProcessor);
+        rep.fenceThreadPool = fenceThreadPool;
+        rep.throttleReadResponses = throttleReadResponses;
+        requestProcessor.onReadRequestStart(requestHandler.ctx().channel());
+        return rep;
+    }
+
+    @Override
+    protected ReferenceCounted readData() throws Exception {
+        ByteBufList data = null;
+        BatchedReadRequest batchRequest = (BatchedReadRequest) request;
+        long maxSize = batchRequest.getMaxSize();
```

Comment:
We need to calculate the maxSize based on `min(nettyMaxFrameSizeBytes - (36 + 4 * batchRequest.getMaxCount()), batchRequest.getMaxSize())`

Reply:
We should let the server decide the frame size. Because you never know how the client set it. If a client sets it too huge, the server will read a lot of entries and make it down.